mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/jmap/JMAPVacationTestSmtpTest.java

```diff
@@ -0,0 +1,137 @@
+
+    public static final String USER = "bob";
+    public static final String DOMAIN = "mydomain.tld";
+    public static final String USER_AT_DOMAIN = USER + "@" + DOMAIN;
+    public static final String PASSWORD = "secret";
+    public static final String HTTP_LOCALHOST = "http://127.0.0.1";
+    public static final String YOPMAIL_COM = "yopmail.com";
+    public static final String MYDOMAIN_TLD = "mydomain.tld";
+    public static final String LOCALHOST = "127.0.0.1";
+    public static final String REASON = "Message explaining my wonderful vacations";
+
+    private final TemporaryFolder folder = new TemporaryFolder();
+    private final GenericContainer fakeSmtp = new GenericContainer("weave/rest-smtp-sink:latest");
+
+    @Rule
+    public final RuleChain chain = RuleChain.outerRule(folder).around(fakeSmtp);
+
+    @Inject
+    private static JmapSmtpHostSystem hostSystem;
+    private int restSMTPPort;
+    private String restSMTPBasedUPI;
+
+    public JMAPVacationTestSmtpTest() throws Exception {
+        super(hostSystem, USER_AT_DOMAIN, PASSWORD, "/org/apache/james/smtp/scripts/");
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        InetAddress containerIp = InetAddresses.forString(fakeSmtp.getContainerInfo().getNetworkSettings().getIpAddress());
+        hostSystem.getInMemoryDnsService()
+            .registerRecord(YOPMAIL_COM, new InetAddress[]{containerIp}, ImmutableList.of(YOPMAIL_COM), ImmutableList.of());
+        hostSystem.getInMemoryDnsService()
+            .registerRecord(DOMAIN, new InetAddress[]{InetAddresses.forString(LOCALHOST)}, ImmutableList.of(MYDOMAIN_TLD), ImmutableList.of());
+
+        restSMTPPort = 80;
+        restSMTPBasedUPI = "http://" + containerIp.getHostAddress();
+        RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
+    }
+
+    @Test
+    public void jmapVacationShouldWork() throws Exception {
+        String bodyRequest = "[[" +
+            "\"setVacationResponse\", " +
+            "{" +
+                "\"update\":{" +
+                    "\"singleton\" : {" +
+                        "\"id\": \"singleton\"," +
+                        "\"isEnabled\": \"true\"," +
+                        "\"textBody\": \"" + REASON + "\"" +
+                    "}" +
+                "}" +
+            "}, \"#0\"" +
+            "]]";
+
+        RestAssured.port = hostSystem.getJmapPort();
+        RestAssured.baseURI = HTTP_LOCALHOST;
+
+        AccessToken accessToken = JmapAuthentication.authenticateJamesUser(USER_AT_DOMAIN, PASSWORD);
+
+        given()
+            .accept(ContentType.JSON)
+            .contentType(ContentType.JSON)
+            .header("Authorization", accessToken.serialize())
+            .body(bodyRequest)
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200);
+
+        scriptTest("helo", Locale.US);
+
+        Thread.sleep(1000);
```

Comment:
Sleep for mail processing. Unfortunately (?) , the mail queue is an asynchronous process... Hence we should wait for the mail being processed.